### PR TITLE
Attack cost additions for fast and slow monsters

### PR DESCRIPTION
**Pull request by SenpaiSlime**
#### Summary
Category "Balance"

#### Purpose of change
Currently, Enemies that move very fast or very slow create unique issues, for fast enemies it makes them extremely dangerous, as attack costs default to 100 and their speed can be upwards of 200, making them able to attack players multiple times a turn. For slow monsters, they are left unintentionally weak, being unable to attack even if you are within range for several turns.

#### Describe the solution
I have gone through the JSON for the monsters in game and have mirrored very slow and very fast enemies attack cost to their speed, this will limit them to a much more reasonable attack rate, I have also gone and done a pass at some other monsters that could use a bit of a lift or tweak, hopefully this will bring everything closer together in how they should behave in combat, and better match their peers

#### Describe alternatives you've considered
Alternatively we could revisit how monster attacking works and make a more complex solution to consider more factors.

#### Testing
I have checked the combat speeds and the fast enemies are now attacking their intended speed, unfortunately slow enemies are at the whims of the game's handling of turns, they still move slow and cant attack while moving, but if you remain in their range their attack speed manages to put them in a real range of threat now

#### Additional context
These numbers are not perfect, and I fully welcome changes in the future, I just wanted to quash what I could for the sake of a better game experience against very fast enemies.

### CODE REVIEW

File: data/json/monsters/zed_misc.json [hunk 1]
```diff
@@ -211,6 +213,7 @@
     "weight": "200 kg",
     "hp": 300,
     "speed": 240,
+    "attack_cost": 240,
```
Comment (kevingranade) [inline data/json/monsters/zed_misc.json:216]: The description specifically calls out that it's overall very fast, not just at movement.
```suggestion
    "attack_cost": 120,
```

**Comment by kevingranade** [comment]
I've spot checked a few things, and while I agree with your general direction, it's clear from the description of some of these monsters that they are intended to be fast at both moving and attacking. This change needs to be a little more cognizant of how the stats match up with the descriptions.  I also noticed some of the changes step outside the bounds of what you outlined, e.g. setting a slower attack speed for a monster with a normal base speed stat.

File: data/json/monsters/nether.json [hunk 2]
```diff
@@ -357,6 +361,7 @@
     "weight": "200 kg",
     "hp": 350,
     "speed": 280,
+    "attack_cost": 280,
```
Comment (Venera3) [inline data/json/monsters/nether.json:364]: Speed = attack cost pretty much neuters any hit-and-run critter (even more than the flag already does, it's a net nerf in 99% of cases).

File: data/json/monsters/mutant.json [hunk 3]
```diff
@@ -110,6 +110,7 @@
     "morale": 13,
     "regen_morale": true,
     "speed": 150,
+    "attack_cost": 130,
```
Comment (Venera3) [inline data/json/monsters/mutant.json:113]: They're designed around being able to move in and attack on the same turn consistently.
```suggestion
```

File: data/json/monsters/insect_spider.json [hunk 4]
```diff
@@ -2848,7 +2854,7 @@
     "name": { "str": "titan stag beetle" },
     "description": "This stag beetle rivals an armored personnel carrier in size, bearing horns that could fell a tree and chitin thick enough to ignore most bullets.  Its bulky shell looks nigh-impenetrable, and the cracks have sealed with a foul paste.",
     "copy-from": "mon_stag_beetle_giant",
-    "proportional": { "hp": 4, "speed": 0.75, "morale": 3, "melee_dice_sides": 2 },
+    "proportional": { "hp": 4, "speed": 0.75, "attack_cost": 0.75, "morale": 3, "melee_dice_sides": 2 },
```
Comment (Venera3) [inline data/json/monsters/insect_spider.json:2857]: This thing is gigantinormous and pretty much invincible,, it doesn't need to put an attack out every turn to be scary.

**Review by Venera3 — COMMENTED** [review]
I agree broad strokes with the intent, but you're missing some context about how the mobs perform. I'd especially reconsider going 1 turn = 1 attack for hit and run monsters, the poor things spend 80% of any given combat fleeing already.

**Comment by kevingranade** [comment]
Also you might want to browse around in your github account and see if any weird messages pop up, otherwise I don't know why the tests aren't running.

File: data/json/monsters/zed_fusion.json [hunk 5]
```diff
@@ -299,6 +299,7 @@
     "weight": "81500 g",
     "hp": 10,
     "speed": 250,
+    "attack_cost": 200,
```
Comment (John-Candlebury) [inline data/json/monsters/zed_fusion.json:302]: Should undo this one too. Scissorlimbs are inteded to be this fast.

For reference they spawn only within a single end-game location, and are designed to energize  with the rest of the collapsed tower zombies They are the high damage dealing unit  that the rest of the crawlers are designed to screen and slow you for.

This nerf makes the whole monsterline much weaker.

**Comment by SenpaiSlime** [comment]
I am going to try and run a V2 of this concept, but narrower scope. Likely just a more in depth look at specific offenders, than a more nuanced balance pass.

**Pull request closed without merging**